fortnite_api/material.py

```diff
@@ -0,0 +1,140 @@
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+from __future__ import annotations
+
+from typing import Any, Dict, Generic, Optional, Tuple
+
+from .abc import Hashable
+from .asset import Asset
+from .enums import CosmeticCompatibleMode
+from .http import HTTPClientT
+from .utils import get_with_fallback
+
+__all__: Tuple[str, ...] = ('MaterialInstanceImages', 'MaterialInstanceColors', 'MaterialInstance')
+
+
+class MaterialInstanceImages(Generic[HTTPClientT]):
+    """
+    .. attributetable:: fortnite_api.MaterialInstanceImages
+
+    Represents some images of a Material instance, as they are rendered in game.
+
+    Attributes
+    ----------
+    offer_image: :class:`fortnite_api.Asset`
+        The offer image of the Material instance. This is the image that is shown in the item shop.
+    texture: Optional[:class:`fortnite_api.Asset`]
+        The texture of the Material instance. This is the texture of the material instance.
+    background: :class:`fortnite_api.Asset`
+        The background of the Material instance. This is the background gradient of the material instance.
+    """
+
+    __slots__: Tuple[str, ...] = ('offer_image', 'background', 'texture')
+
+    def __init__(self, *, data: Dict[str, Any], http: HTTPClientT) -> None:
+        self.offer_image: Asset[HTTPClientT] = Asset(url=data['OfferImage'], http=http)
+
+        _texture = data.get('Texture')
+        self.texture: Optional[Asset[HTTPClientT]] = _texture and Asset(url=_texture, http=http)
+
+        _background = data.get('Background')
+        self.background: Optional[Asset[HTTPClientT]] = _background and Asset(url=_background, http=http)
+
+
+class MaterialInstanceColors:
+    """
+    .. attributetable:: fortnite_api.MaterialInstanceColors
+
+    Represents some metadata about the colors of a Material instance. Every material instance image
+    has a background gradient, made up of a background color A and background color B that falls off to a
+    specific color. This class represents that information.
+
+    Attributes
+    ----------
+    background_color_a: Optional[:class:`str`]
+        The first background color of the Material instance, if any.
+    background_color_b: Optional[:class:`str`]
+        The second background color of the Material instance, if any.
+    fall_off_color: Optional[:class:`str`]
+        The fall off color of the Material instance, if any.
+    """
+
+    __slots__: Tuple[str, ...] = ('background_color_a', 'background_color_b', 'fall_off_color')
+
+    def __init__(self, *, data: Dict[str, Any]) -> None:
+        self.background_color_a: Optional[str] = data.get('Background_Color_A')
+        self.background_color_b: Optional[str] = data.get('Background_Color_B')
+        self.fall_off_color: Optional[str] = data.get('FallOff_Color')
```

Comment:
Here we also got dynamic keys. Here we should also use a similar approach to `MaterialInstanceImages`